pkg/api/replication_test.go

```diff
@@ -11,6 +11,314 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
 package api
 
-// TODO add unit test
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/astaxie/beego"
+	"github.com/astaxie/beego/context"
+	c "github.com/opensds/opensds/pkg/context"
+	"github.com/opensds/opensds/pkg/db"
+	"github.com/opensds/opensds/pkg/model"
+	dbtest "github.com/opensds/opensds/testutils/db/testing"
+)
+
+func init() {
+	beego.Router("/v1beta/block/replications", NewReplicationPortal(),
+		"post:CreateReplication;get:ListReplications")
+	beego.Router("/v1beta/block/replications/detail", NewReplicationPortal(),
+		"get:ListReplicationsDetail")
+	beego.Router("/v1beta/block/replications/:replicationId", NewReplicationPortal(),
+		"get:GetReplication;put:UpdateReplication;delete:DeleteReplication")
+	beego.Router("/v1beta/replications/:replicationId/action", NewReplicationPortal(),
+		"post:Action")
+}
+
+var (
+	fakeReplication = &model.ReplicationSpec{
+		BaseModel: &model.BaseModel{
+			Id:        "f4a5e666-c669-4c64-a2a1-8f9ecd560c78",
+			CreatedAt: "2017-10-24T16:21:32",
+		},
+		Name:              "fake replication",
+		Description:       "fake replication",
+		AvailabilityZone:  "default",
+		PrimaryVolumeId:   "d3a109ff-3e51-4625-9054-32604c79fa90",
+		SecondaryVolumeId: "d3a109ff-3e51-4625-9054-32604c79fa92",
+		ReplicationMode:   "sync",
```

Comment:
IMO, there should be some standard definition about `ReplicationMode`, you can add `sync` and `async` option now, but it could be changed after the profile design proposal finished.

Reply:
fine